nautobot/extras/context_managers.py

```diff
@@ -116,13 +121,34 @@ def web_request_context(user, context_detail="", change_id=None):
     :param user: User object
     :param context_detail: Optional extra details about the transaction (ex: the plugin name that initiated the change)
     :param change_id: Optional uuid object to uniquely identify the transaction. One will be generated if not supplied
+    :param context: Optional string value of the generated change log entries' "change_context" field, defaults to ObjectChangeEventContextChoices.CONTEXT_ORM.
+        Valid choices are in nautobot.extras.choices.ObjectChangeEventContextChoices
+    :param request: Optional web request instance, one will be generated if not supplied
     """
+    from nautobot.extras.jobs import enqueue_job_hooks  # prevent circular import
 
-    if not isinstance(user, get_user_model()):
-        raise TypeError("The user object must be an instance of nautobot.users.models.User")
+    valid_contexts = {
+        ObjectChangeEventContextChoices.CONTEXT_JOB: JobChangeContext,
+        ObjectChangeEventContextChoices.CONTEXT_JOB_HOOK: JobHookChangeContext,
+        ObjectChangeEventContextChoices.CONTEXT_ORM: ORMChangeContext,
+        ObjectChangeEventContextChoices.CONTEXT_WEB: WebChangeContext,
+    }
 
-    request = RequestFactory().request(SERVER_NAME="web_request_context")
-    request.user = user
-    change_context = ORMChangeContext(request=request, context_detail=context_detail, change_id=change_id)
-    with change_logging(change_context):
-        yield request
+    if context not in valid_contexts:
+        raise TypeError(f"{context} is not a valid context")
+
+    if not isinstance(user, (get_user_model(), AnonymousUser)):
```

Comment:
We're not but since every web request is being funneled through this we have to account for AnonymousUser. We could exit early if the user is AnonymousUser but from a security perspective I'd rather the change be logged with user=None